src/editors/codeedit.cpp

```diff
@@ -2853,5 +2853,29 @@ void CodeEdit::moveLines(bool moveLinesUp)
         mCompleter->suppressNextOpenTrigger();
 }
 
+void CodeEdit::trimtrailing()
+{
+    QTextCursor c= textCursor();
+    c.movePosition(c.Start);
+    c.movePosition(c.EndOfLine);
+    for(int i=0;i<blockCount();i++){
```

Comment:
blockCount() here iterates over the whole document. You probably wanted to iterate over the selection: c.block() gives the block of the position. For the anchor you can create a second QTextCursor, set it's position to c.anchor() and get the block of it.